src/definitions.jl

```diff
@@ -271,7 +271,15 @@ summary(p::ScaledPlan) = string(p.scale, " * ", summary(p.p))
 *(p::Plan, I::UniformScaling) = ScaledPlan(p, I.λ)
 
 # Normalization for ifft, given unscaled bfft, is 1/prod(dimensions)
-normalization(::Type{T}, sz, region) where T = one(T) / Int(prod(sz[r] for r in region))::Int
+# ensure that region is a subset of eachindex(sz).
+_checkindex(szinds, region::AbstractVector) = checkindex(Bool, szinds, region)
+# this method handles the case where region is not an array, e.g. it is a Tuple
+_checkindex(szinds, region) = all(r -> checkindex(Bool, szinds, r), region)
+@inline function normalization(::Type{T}, sz, region) where T
+    @boundscheck (!isempty(region) && _checkindex(eachindex(sz), region)) ||
```

Comment:
Why do we want to require `!isempty(region)` here?

Reply:
This seems to be to disallow calls like
```julia
julia> AbstractFFTs.normalization(Float64, (2,), Int[])
1.0
```
and instead require `region to correspond to the dimensions. I'm not certain if allowing the empty case makes an actual difference, as a normalization factor of `1` wouldn't change anything.